flink-clients/src/main/java/org/apache/flink/client/program/MiniClusterClient.java

```diff
@@ -61,7 +61,10 @@ public JobSubmissionResult submitJob(JobGraph jobGraph, ClassLoader classLoader)
 
 		if (isDetached()) {
 			try {
-				return jobSubmissionResultFuture.get();
+				final JobSubmissionResult jobSubmissionResult = jobSubmissionResultFuture.get();
+
+				this.lastJobExecutionResult = new DetachedJobExecutionResult(jobSubmissionResult.getJobID());
```

Comment:
Nice catch! I just notice that we not set `lastJobExecutionResult` in MiniCluster case. I think it might be hopefully we eventually get rid of it(we will unify all submission code path by `submitJob` instead of `run`s which the only method require `lastJobExecutionResult`), but let's keep consistency now which also helps convinced refactors.

Reply:
Yes, I fixed that in this PR. Before it was not failing because in the `ClusterClient.run(...)` we had the second check in the `if (lastJobExecutionResult == null) ...` which was returning `false`.